frontend/src/profile-registration/GeneralInfoSection.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react'
+import EmailInput from './EmailInput'
+import NameInput from './NameInput'
+import UsernameInput from './UsernameInput'
+
+export const DEFAULT_GENERAL_INFO = {
+    name: "",
+    username: "",
+    email: ""
+}
+
+export class GeneralInfoSection extends Component {
+
+    constructor(props) {
+        super(props)
+        this.state = { generalInfo: DEFAULT_GENERAL_INFO }
+    }
+
+    handleNameChange(name) {
+        this.setState(
+            s => ({ generalInfo: { ...s.generalInfo, name } }),
```

Comment:
if you use plain state like 
```
{
    name: "",
    username: "",
    email: ""
}
```
you can update state property
{ name }
instead of copying { generalInfo: { ...s.generalInfo, name } }

Reply:
@dmitrychernykh It's a good note, that I can simply use `setState({ name })`, because `setState(...)` merges the previous value with the new one.